### PR TITLE
feat: Unpick items in My List

### LINKED ISSUE #1566: Make the item disappear from the MyLists page when it is un-saved.
## 🎉 Description

* [ ] 🐛 This is a bug report.
* [x] 📈 This is a feature request.

Currently, the item stays there until the page is refreshed.

Note: if I un-save it, the item disappears, but if I click on UNDO in the toast, it should appear again without the need to reload the page.

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the capability of unpicking items in the My List section.
Closes #1566

### CODE REVIEW

File: webapp/src/modules/ui/browse/selectors.ts [hunk 1]
```diff
@@ -60,6 +64,40 @@ export const getOnSaleItems = createSelector<
...
+    .filter(item => favoritedItems[item.id]?.pickedByUser === true)
+    .sort((a, b) => {
+      const favoriteACreatedAt = favoritedItems[a.id]?.createdAt ?? 0
+      const favoriteBCreatedAt = favoritedItems[b.id]?.createdAt ?? 0
+      if (favoriteACreatedAt < favoriteBCreatedAt) {
+        return 1
+      } else if (favoriteACreatedAt > favoriteBCreatedAt) {
+        return -1
+      }
+      return 0
```
Comment (kevinszuchet) [inline webapp/src/modules/ui/browse/selectors.ts:97]: WDTY about moving this logic into a new function `byCreatedAtAsc` (or something similar), so the selector could be more readable?

**Review by kevinszuchet — COMMENTED** [review]
Great job! I like the solution. 👏👏👏

I've left some comments

File: webapp/src/modules/favorites/selectors.spec.ts [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
   getError,
   getFavoritedItems,
+  isLoadingFavoritedItems,
   getFavoritesDataByItemId,
   getIsPickedByUser,
   getListId,
   getLoading,
   getState
 } from './selectors'
 import { locations } from '../routing/locations'
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/modules/favorites/selectors.spec.ts]: Done!

File: webapp/src/modules/favorites/selectors.ts [hunk 3]
```diff
@@ -1,15 +1,20 @@
 import { createMatchSelector } from 'connected-react-router'
+import { isLoadingType } from 'decentraland-dapps/dist/modules/loading/selectors'
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/modules/favorites/selectors.ts]: Done!

File: webapp/src/modules/ui/browse/reducer.ts [hunk 4]
```diff
@@ -147,6 +165,7 @@ export function browseReducer(
       }
 
     case FETCH_FAVORITED_ITEMS_SUCCESS:
+      console.log('Total success items', action.payload.total)
```
[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply webapp/src/modules/ui/browse/reducer.ts]: Done!

File: webapp/src/modules/ui/browse/utils.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+) => (a: Item, b: Item) => {
+  const favoriteACreatedAt = favoritedItems[a.id]?.createdAt ?? 0
+  const favoriteBCreatedAt = favoritedItems[b.id]?.createdAt ?? 0
+  if (favoriteACreatedAt < favoriteBCreatedAt) {
+    return 1
+  } else if (favoriteACreatedAt > favoriteBCreatedAt) {
+    return -1
+  }
+  return 0
+}
```
Comment (kevinszuchet) [inline webapp/src/modules/ui/browse/utils.ts:20]: It would be great to add tests to this file.

**Review by kevinszuchet — DISMISSED** [review]

**Review by kevinszuchet — APPROVED** [review]
❤️

**Pull request merged**